website/client/components/challenges/challengeDetail.vue

```diff
@@ -365,32 +365,32 @@ export default {
       this.$store.state.memberModalOptions.groupId = 'challenge'; // @TODO: change these terrible settings
       this.$store.state.memberModalOptions.group = this.group;
       this.$store.state.memberModalOptions.viewingMembers = this.members;
-      this.$root.$emit('bv::show::modal', 'members-modal');
+      this.$root.$emit('bv::show::modal', 'members-modal', 'bv::hide::modal');
     },
     async joinChallenge () {
       this.user.challenges.push(this.searchId);
       await this.$store.dispatch('challenges:joinChallenge', {challengeId: this.searchId});
       await this.$store.dispatch('tasks:fetchUserTasks', {forceLoad: true});
     },
     async leaveChallenge () {
-      this.$root.$emit('bv::show::modal', 'leave-challenge-modal');
+      this.$root.$emit('bv::show::modal', 'leave-challenge-modal', 'bv::hide::modal');
     },
     closeChallenge () {
-      this.$root.$emit('bv::show::modal', 'close-challenge-modal');
+      this.$root.$emit('bv::show::modal', 'close-challenge-modal', 'bv::hide::modal');
     },
     edit () {
       // @TODO: set working challenge
       this.cloning = false;
       this.$store.state.challengeOptions.workingChallenge = Object.assign({}, this.$store.state.challengeOptions.workingChallenge, this.challenge);
-      this.$root.$emit('bv::show::modal', 'challenge-modal');
+      this.$root.$emit('bv::show::modal', 'challenge-modal', 'bv::hide::modal');
     },
     // @TODO: view members
     updatedChallenge (eventData) {
       Object.assign(this.challenge, eventData.challenge);
     },
     openMemberProgressModal (memberId) {
       this.progressMemberId = memberId;
-      this.$root.$emit('bv::show::modal', 'challenge-member-modal');
+      this.$root.$emit('bv::show::modal', 'challenge-member-modal', 'bv::hide::modal');
```

Comment:
Can we add a comment above these calls to explain what's happening and why the third argument is necessary? Have you figured out why this is happening only with certain modals?

@TheHollidayInn you might want to give a look at this

Reply:
@paglias, unfortunately I don't have that much of experience to figure out why some modals appear randomly. Could it be caused by the recent bootstrap-vue upgrade? 

Do you think this have anything to do with this issue? 
[ **Auto Return Focus**
When < b-modal > is opened via the < v-b-modal > directive on an element, focus will be returned to this element automatically when < b-modal > closes, unless an element has been specified via the return-focus prop.](https://bootstrap-vue.js.org/docs/components/modal/)

@Alys suggested we include the fix for issue #9531 with this PR. I'll wait for @TheHollidayInn to review this first. Thanks.